### DISCUSSION TITLE
IOT features request

### DISCUSSION THREAD

**Discussion by liqijian101** [luc-github/ESP3D-TFT#65] [Ideas]
Are we preparing to develop IoT functionality?
If equipped with IoT functionality, 3D printers can be remotely controlled.
We can use free IoT servers, such as Blinker.

https://github.com/blinker-iot/blinker-esp-idf

Just implementing simple control is also acceptable。 For example, starting printing, stopping printing,

**Comment by luc-github** [comment]
I do not have any user case for this so far.

What do you mean by ` IoT functionality` ? do you refere to generic term or to a specific API ?

**Comment by liqijian101** [comment]
The Internet of things
For example,   Google Cloud IoT

**Comment by luc-github** [comment]
That mean nothing - it is a concept of communication not a precise API - it is like MQTTT - it is just a communication protocol but not a clear API of what message / format of message / data,  must be used and transmisted using this protocol for a specific purpose 
it is very vague

**Comment by liqijian101** [comment]
Yes, it uses the Mqtt communication protocol. What we need is not this protocol, we need the ability to remotely operate the printer. So we can use this protocol and a free server to implement it.

**Reply by luc-github** [reply]
I am not expert on this so I won't comment more - but to be honest  it is unlikely that I will spend time even thinking on this before the already planned features are done

**Comment by liqijian101** [comment]
Okay, I will first improve my motherboard and then try to add this feature

**Reply by luc-github** [reply]
if you can specify it clearily it would definitly be a first step

**Comment by 3DSmitty** [comment]
> Yes, it uses the Mqtt communication protocol. What we need is not this protocol, we need the ability to remotely operate the printer. So we can use this protocol and a free server to implement it.

You can already remotely operate the printer through the web interface. The only advantage I can see from Mqtt is if you have a print farm and need to control / track many printers on one web page.
